sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -355,6 +356,7 @@ public synchronized BigQueryStorageStreamSource<T> getCurrentSource() {
                       .build(),
                   source.readSession.getTable());
           newResponseIterator = newResponseStream.iterator();
+          newResponseIterator.hasNext();
```

Comment:
Probably let's add a log line here describing why this call is important. For someone who doesn't understand the details regarding the BQ API, it's just an ignored hasNext call which might be removed in the future cleanups.

Also, does this points to an issue with the API ?

Reply:
Added a comment.

I don't think this indicates an issue with the API per se. We could probably try and read from the stream more explicitly in case that is clearer but the comment also points out why this important now.

Reply:
Would it be feasible to add a test that exercises this behavior?